File: back-end-lims/master-data-service/src/main/java/com/digiworldexpo/lims/master/service/ReportDateFormatService.java
package com.digiworldexpo.lims.master.service;

import java.util.List;
import java.util.UUID;

import com.digiworldexpo.lims.entities.master.ReportDateFormat;
import com.digiworldexpo.lims.master.model.response.ResponseModel;

public interface ReportDateFormatService {
	
	ResponseModel<ReportDateFormat> saveReportDateFormat(ReportDateFormat reportDateFormat, UUID createdBy);

	ResponseModel<List<ReportDateFormat>> getReportDateFormats(String startsWith, int pageNumber, int pageSize,
			String sortBy, UUID createdBy);

	ResponseModel<ReportDateFormat> getReportDateFormatById(UUID id);

	ResponseModel<ReportDateFormat> updateReportDateFormat(UUID id, ReportDateFormat updatedReportDateFormat, UUID userId);

	ResponseModel<ReportDateFormat> deleteReportDateFormat(UUID id);


}

File: lims-web-app/src/services/MasterDataService/ReportSettingsMaster/ReportDateFormatService.js
import { Urls } from '../../apiURLS';
import axios from 'axios';

const BASE_URL = Urls.MASTERS;

export const getAllReportDateFormats = async (startsWith = '', pageNumber = 0, pageSize = 10, sortBy = 'format', createdBy = '') => {
    const response = await axios.post(
        `${BASE_URL}report-date-format/get-all?startsWith=${startsWith}&pageNumber=${pageNumber}&pageSize=${pageSize}&sortBy=${sortBy}`, 
        {}, 
        { headers: { 'createdBy': createdBy } }
    );
    return response.data;
};

export const saveReportDateFormat = async (reportDateFormat, createdBy) => {
    const response = await axios.post(
        `${BASE_URL}report-date-format/save`, 
        reportDateFormat, 
        { headers: { 'createdBy': createdBy } }
    );
    return response.data;
};

export const deleteReportDateFormatById = async (id) => {
    const response = await axios.delete(`${BASE_URL}report-date-format/${id}`);
    return response.data;
};

export const getReportDateFormatById = async (id) => {
    const response = await axios.get(`${BASE_URL}report-date-format/${id}`);
    return response.data;
};

export const updateReportDateFormat = async (id, updatedReportDateFormat, userId) => {
    const response = await axios.put(
        `${BASE_URL}report-date-format/update/${id}`, 
        updatedReportDateFormat, 
        { headers: { 'userId': userId } }
    );
    return response.data;
};

export const reportDateFormatService = {
    getAllReportDateFormats,
    saveReportDateFormat,
    deleteReportDateFormatById,
    getReportDateFormatById,
    updateReportDateFormat
};
